### PR TITLE
Add BGP route APIs

### PR DESCRIPTION
**Pull request by lamchester**
Hey @lamchester 

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

Add APIs for BGP peer status, learned routes, and advertised routes for Azure VPN gateways

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@lamchester__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (chelam). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by fearthecowboy** [comment]
Hey @lamchester 

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

File: arm-network/2016-09-01/swagger/network.json [hunk 1]
```diff
@@ -4059,6 +4059,149 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/getBgppeerstatus": {
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: casing intentional?

File: arm-network/2016-09-01/swagger/network.json [hunk 2]
```diff
@@ -4059,6 +4059,149 @@
...
+            }
+          },
+          "202": {
+            "description": ""
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/getLearnedroutes": {
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: casing intentional? kind of inconsistent with next path (`getAdvertisedRoutes`)

File: arm-network/2016-09-01/swagger/network.json [hunk 3]
```diff
@@ -4059,6 +4059,149 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "List of learned BGP routes",
+            "schema": {
+              "$ref": "#/definitions/GatewayRouteListResult"
+            }
+          },
+          "202": {
+            "description": ""
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: ...

File: arm-network/2016-09-01/swagger/network.json [hunk 4]
```diff
@@ -6469,6 +6612,79 @@
         }
       }
     },
+    "BgpPeerStatus": {
+      "properties": {
+        "localAddress": {
+          "type": "string",
+          "description": "Gets or sets the virtual network gateway's local address"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: >Gets or sets [](start = 26, length = 12)

Please remove "Gets or sets" from descriptions - the language generator will make appropriate text regarding property accessors. (applies to all other descriptions)

File: arm-network/2016-09-01/swagger/network.json [hunk 5]
```diff
@@ -6469,6 +6612,79 @@
...
+          "description": "Gets or sets the remote BGP peer"
+        },
+        "asn": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the autonomous systen number of the remote BGP peer"
+        },
+        "state": {
+          "type": "string",
+          "description": "Gets or sets the BGP peer state (unknown, stopped, idle, connecting, connected)"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: if the set of values is known, make this an enum.

File: arm-network/2016-09-01/swagger/network.json [hunk 6]
```diff
@@ -6469,6 +6612,79 @@
...
+        },
+        "connectedDuration": {
+          "type": "string",
+          "description": "Gets or sets how long the peering has been up"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: this is a string? what are the possible values?

File: arm-network/2016-09-01/swagger/network.json [hunk 7]
```diff
@@ -6469,6 +6612,79 @@
         }
       }
     },
+    "BgpPeerStatus": {
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: This model is used in responses only (right?) so there is no point for users to modify the properties. Mark them as `readOnly` then.

File: arm-network/2016-09-01/swagger/network.json [hunk 8]
```diff
@@ -6469,6 +6612,79 @@
...
+          "type": "string",
+          "description": "Gets or sets the gateway's local address"
+        },
+        "network": {
+          "type": "string",
+          "description": "Gets or sets the route's network prefix"
+        },
+        "nextHop": {
+          "type": "string",
+          "description": "Gets or sets the route's next hop"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: I'm not an expert on this topic but maybe users are neither: Is this an IP address or something else? Maybe make the description more verbose

File: arm-network/2016-09-01/swagger/network.json [hunk 9]
```diff
@@ -4059,6 +4059,149 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of BGP peer statuses",
+            "schema": {
+              "$ref": "#/definitions/BgpPeerStatusListResult"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: Use `x-ms-pageable` to improve the user experience of this return type. See the other operations in this file ;-)

File: arm-network/2016-09-01/swagger/network.json [hunk 10]
```diff
@@ -4059,6 +4059,149 @@
...
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of learned BGP routes",
+            "schema": {
+              "$ref": "#/definitions/GatewayRouteListResult"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: Use `x-ms-pageable` to improve the user experience of this return type.

File: arm-network/2016-09-01/swagger/network.json [hunk 11]
```diff
@@ -8220,6 +8436,38 @@
...
       "description": "Response for the ListVirtualNetworkGateways API service call."
     },
+    "BgpPeerStatusListResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BgpPeerStatus"
+          },
+          "description": "Gets list of BGP peers"
```
Comment (olydis) [inline arm-network/2016-09-01/swagger/network.json]: Just another quick reminder: No "Gets" and such. Note that this also affects descriptions that are not part of this commit, please fix them in the same run :)

**Review by olydis — CHANGES_REQUESTED** [review]
Please address comments.

**Comment by olydis** [comment]
>     "AzureAsyncOperationResult": {

are you aware that this type is never used?

---
Refers to: arm-network/2016-09-01/swagger/network.json:8611 in 3a89cd3. [](commit_id = 3a89cd30c4a8b5f219e09ce723534ffbadbd8e95, deletion_comment = False)

**Comment by olydis** [comment]
You have checked "Validation errors from the Linter extension for VS Code have all been fixed for this spec." - this is **not** the case, there are still countless messages.

**Review by olydis — APPROVED** [review]

**Review by olydis — CHANGES_REQUESTED** [review]
mistakenly approved

**Comment by lmazuel** [comment]
Hi @lamchester 
I'm not in the Swagger team but the Python team. However, I think I'm correct to tell you that the Network team has split the network files into small files and now use Composite mode. [The change is already made in C#](https://github.com/Azure/azure-sdk-for-net/pull/2660), so the file you change in this PR is not used anymore.
You should consider open a new PR against the correct sub-file.

**Comment by lamchester** [comment]
Network spec has been split up, going to open another pull request.
